src/keenio.rs

```diff
@@ -82,33 +106,33 @@ impl KeenClient {
         }
     }
 
-    pub fn flush(&mut self, wait: bool) -> Result<(), String> {
+    pub fn flush(&mut self, wait: bool) -> Result<(), Error> {
         // Send the event FLUSH
         let sender = self.sender.lock().unwrap();
 
         if let Some(ref sender) = *sender {
-            sender.send(Event::Flush(wait)).map_err(|e| e.to_string())?;
+            sender.send(Event::Flush(wait)).map_err(|e| Error::Io(e.to_string()))?;
```

Comment:
We could implement From<Io::Error> to Error and keep the io error in the Error instead of the string.  And here, we could use the ? notation without a map_err.

Reply:
I agree in principle but we have to wrap both `mpsc::SendError` and `mpsc::RecvError`; so e would need something like `IoSend` and `IoRecv` in our error type. Or maybe there is a better way. What do you think?

Reply:
Oh, I though it was an Io::Error, but it is a mpsc::SendError and a mpsc::RecvError. My bad, that's ok like this 👍

Reply:
A more thorough approach would be better. This can be an incremental improvement for now :) Thanks!